cmd/skopeo/proxy.go

```diff
@@ -266,6 +268,23 @@ func (h *proxyHandler) openImageImpl(args []any, allowNotFound bool) (replyBuf,
 		return ret, err
 	}
 
+	unparsedTopLevel := image.UnparsedInstance(imgsrc, nil)
+	policy, err := signature.DefaultPolicy(h.sysctx)
```

Comment:
Was there a specific reason not to use `opts.global.getPolicyContext` here?

That would also immediately handle `--insecure-policy` per https://github.com/containers/skopeo/pull/2029#pullrequestreview-1506680929 .